opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -675,8 +659,45 @@ boolean hasFileBasedTags() {
         return true;
     }
 
+    /**
+     * @param dotGit {@code .git} file
+     * @return value of the {@code gitdir} property from the file
+     */
+    private String getGitDirValue(File dotGit) {
+        try (Scanner scanner = new Scanner(dotGit, StandardCharsets.UTF_8)) {
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                if (line.startsWith(Constants.GITDIR)) {
+                    return line.substring(Constants.GITDIR.length());
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "failed to scan the contents of file ''{0}''", dotGit);
+        }
+
+        return null;
+    }
+
     private org.eclipse.jgit.lib.Repository getJGitRepository(String directory) throws IOException {
-        return FileRepositoryBuilder.create(Paths.get(directory, ".git").toFile());
+        File dotGitFile = Paths.get(directory, Constants.DOT_GIT).toFile();
+        if (dotGitFile.isDirectory()) {
+            return FileRepositoryBuilder.create(dotGitFile);
+        }
+
+        // Assume this is a sub-module so dotGitFile is a file.
+        String gitDirValue;
+        if ((gitDirValue = getGitDirValue(dotGitFile)) == null) {
```

Comment:
Just a matter of taste but I find this cleaner:
```java
String gitDirValue = getGitDirValue(dotGitFile);
if (gitDirValue == null) {
```

Reply:
Yep. Strange how I find this perfectly acceptable in C code and in Java code this looks a bit awkward.